### DISCUSSION TITLE
Documentation for assembly output format from "emcc -S"?

### DISCUSSION THREAD

**Discussion by gregbuchholz** [emscripten-core/emscripten#15929] [Q&A] [answered]
I'm wondering where I can find the documentation describing the assembly output format used for the files generated by using the "-S" format to emcc.  For instance, if you take [example.c](https://github.com/gregbuchholz/thread_local_storage/blob/main/src/c_example/example.c) and run `emcc example.c -S -pthread`, you get this output file: [example.s](https://github.com/gregbuchholz/thread_local_storage/blob/main/src/c_example/example.s).  Obviously the assembly instructions are web-assembly, but I'm interested in learning more about what the "T" and @ represent in a directive like:

`.section	.tdata.variable1,"T",@`

..etc.

Thanks!

**Comment by sbc100** [comment, marked as answer]
This is the assembly format that is output by llvm (and read by llvm).  I'm afraid there is no good documentation for this specific format but it shares a lot with standard GNU assembly format used by llvm on other platforms.  For example the "T" there means that section is thread local (TLS).

See https://ftp.gnu.org/old-gnu/Manuals/gas-2.9.1/html_chapter/as_7.html (although I notice that "T" is not listed amount that flags there :).

**Reply by gregbuchholz** [reply]
Thanks for the info.

**Comment by sbc100** [comment]
Is there some reason you are interested in this format?  Do you have plans to author files directly in assembler perhaps?

**Reply by gregbuchholz** [reply]
My interest just comes from as part of a debugging exercise to try and identify the source of the problem outlined [here](https://github.com/emscripten-core/emscripten/issues/15891).  I do not have any plans on authoring wasm files in this format.
